src/screens/MainApp/MyDiagnoses/index.js

```diff
@@ -110,21 +119,23 @@ MyDiagnoses.navigationOptions = () => ({
 const MyDiagnosesStack = buildStackNavigator(
   {
     MyDiagnoses,
-    DetailedDiagnose
+    DetailedDiagnose,
+    SolutionRequest,
+    FinishScreen
   },
   {
     initialRouteName: 'MyDiagnoses'
   }
 )
 
-const allowedRoutesToShowTabBar = ['MyDiagnoses']
+const allowedRoutesToShowTabBar = ['MyDiagnoses', 'FinishScreen']
 
 MyDiagnosesStack.navigationOptions = ({ navigation }) => {
   if (navigation.state.routes.length <= 1) return {}
 
-  const routesInHistoryWithTabBar = intersection(navigation.state.routes, allowedRoutesToShowTabBar)
+  const routesInState = navigation.state.routes.map(route => route.routeName)
+  const tabBarVisible = includes(allowedRoutesToShowTabBar, routesInState.pop())
```

Comment:
The `map` and `pop` are unnecessary. The following code is equivalent and `O(1)`:
```javascript
const routes = navigation.state.routes
const latestRouteName = routes[routes.length - 1].routeName
const tabBarVisible = includes(allowedRoutesToShowTabBar, latestRouteName)
```